src/Responses/GetRecordingTextTracksResponse.php

```diff
@@ -27,23 +27,17 @@
  */
 class GetRecordingTextTracksResponse extends BaseJsonResponse
 {
-    /**
-     * @var Track[]
-     */
-    private $tracks;
-
     /**
      * @return Track[]
      */
-    public function getTracks()
+    public function getTracks(): array
     {
-        if (null === $this->tracks) {
-            $this->tracks = [];
-            foreach ($this->data->response->tracks as $track) {
-                $this->tracks[] = new Track($track);
-            }
+        $tracks = [];
```

Comment:
Trade-off between more memory (old version) and more cpu (new version). Need a good argument to decided which solution to choose for the community. We assume the XML content is static and never changes in a single API call.

Reply:
Technically you are right (memory vs. cpu), but I guess those API-small data do no harm to any of them in these days. This proposal is just a cleaner and leaner way ... and it is more robust against programming errors (less code, less variables, low risk of inconsistent data).